components/modal/src/styled.js

```diff
@@ -0,0 +1,80 @@
+/** @jsx jsx */
+
+import React from 'react';
+import { jsx, useTheme, paint } from '@westpac/core';
+
+export const ModalBackdrop = props => {
+	const styles = {
+		position: 'fixed',
+		zIndex: '1001',
+		backgroundColor: 'rgba(0,0,0,0.5)',
+		top: 0,
+		right: 0,
+		bottom: 0,
+		left: 0,
+		display: 'flex',
+		justifyContent: 'center',
+		alignItems: 'flex-start',
+		transition: 'all 0.3s ease',
+
+		'&.modal-backdrop-enter': {
+			opacity: 0,
+		},
+
+		'&.modal-backdrop-enter-active': {
+			opacity: 1,
+		},
+
+		'&.modal-backdrop-exit': {
+			opacity: 1,
+		},
+
+		'&.modal-backdrop-exit-active': {
+			opacity: 0,
+		},
+	};
+	return <div css={styles} {...props} />;
+};
+
+export const StyledModal = ({ size, ...props }) => {
+	const { modal, breakpoints } = useTheme();
+	const mq = paint(breakpoints);
+
+	const styles = {
+		overflow: 'auto',
```

Comment:
Needs a 12px margin so it doesn't reach the viewport edge on XS. GUI2.0 has a 12px edge gap using left/right values.

Reply:
Also need to be able to scroll modal content. Resize viewport to short height and check with a modal with long content.